### PR TITLE
325 json to csv package

### LINKED ISSUE #325: Implement a complete package for parsing JSON products to CSV
### Description

Implement a module to parse commercetools products from JSON format to CSV

(Part of a series implementing the rewrite of the [product-csv-sync](https://github.com/sphereio/sphere-node-product-csv-sync) module)

#292 is getting too large so will be broken down into smaller issues and smaller PRs, with this issue encompassing all the issues

### TODO

- [x] Implement core functionality(#292)

- [x] Implement CLI wrapper (#332)

- [x] Add integration tests (#332)

- [x] Add documentation

### PR DESCRIPTION
**Pull request by wizzy25**
#### Summary

**The big one is finally here!**

This PR adds a new package that parses product data to CSV

Most of this PR has been reviewed in smaller stages before now so this shouldn't be too daunting

For more context on this, take a look at [this wiki](https://wiki.commercetools.com/display/S/Product+Exporter+Migration)

resolves #325

### CODE REVIEW

**Comment by tdeekens** [comment]
So how much review do you expect on this as all smaller parts have been reviewed?

**Comment by wizzy25** [comment]
@tdeekens  As little/much as possible. It really isn't even all that necessary and can be merged anyway...

File: packages/product-json-to-csv/test/writer.spec.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        'variantId',
+        'sku',
+        'images',
+        'anotherAddedAttr',
+        'article',
+        'designer',
+        'color',
+      ]
+      const outputStream = streamTest.toText((error, actual) => {
+        expect(true).toBeTruthy()
```
Comment (adnasa) [inline packages/product-json-to-csv/test/writer.spec.js]: wait... why is this needed?

Comment (wizzy25) [reply packages/product-json-to-csv/test/writer.spec.js]: oops

**Comment by adnasa** [comment]
👍 
Let me know if this needs another round again. but from what I remember last, everything seemed fine expect for the specific `it` that I found

**Comment by wizzy25** [comment]
I'll try to increase test coverage before I merge 🔢

**Comment by wizzy25** [comment]
Here goes the big one 🚀
